### PR TITLE
Link email fronts to corresponding network fronts

### PR DESCRIPTION
**Pull request by davidfurey**
## What does this change?

Include canonical and special containers from uk/us/au network front to uk/us/au Guardian Today email

## What is the value of this and can you measure success?

Guardian Today emails are automated, and are configured to loosely mirrors the containers on the network front.  If a special or canonical container is added to the network front and isn't copied onto the Guardian Today front there is a risk that the most important story of the day will be missing from the email.

## Does this affect other platforms - Amp, Apps, etc?

No

### CODE REVIEW

File: facia-press/app/frontpress/FapiFrontPress.scala [hunk 1]
```diff
@@ -183,12 +184,41 @@ trait FapiFrontPress extends Logging with ExecutionContexts {
       .map(_.map(PressedContent.make))
   }
 
+  private def isHighPriorityCollection(collection: CollectionConfigJson): Boolean = {
+    collection.metadata.exists(_.exists({
+      case Special | Canonical => true
```
Comment (joelochlann) [inline facia-press/app/frontpress/FapiFrontPress.scala]: I know I really shouldn't be looking at this while on holiday, but I had to open my work email to look for a holiday-related email, so... here I am. Just to say this is the relevant code in the old email renderer, which may be helpful:

- https://github.com/guardian/gu-email-renderer/blob/a91bcf5da07ba4bb511d1e20d007b0e6697a030a/template/us/daily/categories_us.html#L19-L63
- https://github.com/guardian/gu-email-renderer/blob/422446b3c79e43fefe03407f93e1d8bffee5663d/template/uk/daily/categories.html#L19-L43
- https://github.com/guardian/gu-email-renderer/blob/422446b3c79e43fefe03407f93e1d8bffee5663d/template/au/daily/v4.html#L19-L43

File: facia-press/app/frontpress/FapiFrontPress.scala [hunk 2]
```diff
@@ -183,12 +184,41 @@ trait FapiFrontPress extends Logging with ExecutionContexts {
...
+
   private def getCollectionIdsForPath(path: String): Response[List[String]] =
-    for(
-      fronts <- FAPI.getFronts()
-    ) yield fronts.find(_.id == path).map(_.collections).getOrElse {
-      log.warn(s"There are no collections for path $path")
-      throw new IllegalStateException(s"There are no collections for path $path")
+    for {
+      config <- Response.Async.Right(fapiClient.config)
+    } yield {
```
Comment (jfsoul) [inline facia-press/app/frontpress/FapiFrontPress.scala]: Is there value in this being a for-comprehension vs. `Response.Async.Right(fapiClient.config).flatMap { config => ...`?

Comment (davidfurey) [inline facia-press/app/frontpress/FapiFrontPress.scala]: There might have been once, but not anymore.

**Review by jfsoul — APPROVED** [review]
Looks good to me, ~~although I did have a question~~.

File: facia-press/app/frontpress/FapiFrontPress.scala [hunk 3]
```diff
@@ -183,12 +184,42 @@ trait FapiFrontPress extends Logging with ExecutionContexts {
...
   }
 
+  private def collectionWithMetadata(metadata: Metadata, path: String, config: ConfigJson): Option[String] =
+    (for {
+      front <- config.fronts.get(path)
+      collectionId <- front.collections
+      collectionConfig <- config.collections.get(collectionId) if collectionConfig.metadata.exists(_.contains(metadata))
+    } yield collectionId).headOption
+
+  private def withHigPriorityCollections(parentPath: String, config: ConfigJson, collections: List[String]): List[String] =
```
Comment (joelochlann) [inline facia-press/app/frontpress/FapiFrontPress.scala]: typo - should be `withHighPriorityCollections`

File: facia-press/app/frontpress/FapiFrontPress.scala [hunk 4]
```diff
@@ -183,12 +184,42 @@ trait FapiFrontPress extends Logging with ExecutionContexts {
...
+    collections match {
+      case head :: tail =>
+        List(
+          collectionWithMetadata(Breaking, parentPath, config),
+          Some(head),
```
Comment (joelochlann) [inline facia-press/app/frontpress/FapiFrontPress.scala]: Worth noting that this means that for the old email renderer functionality to be replicated exactly, the top container must be backfilled with the canonical container on the corresponding front - which seems fine to me, though worth checking if the canonical container ever changes

**Review by joelochlann — APPROVED** [review]
LGTM 👍 , apart from a few very minor naming things

File: facia-press/app/frontpress/FapiFrontPress.scala (hunk 4 above)
Comment (davidfurey) [inline facia-press/app/frontpress/FapiFrontPress.scala]: Yes.  I think we've both discussed this with Celine and this is okay.

**Pull request merged**